### ISSUE TITLE
IllegalArgumentException Thrown from WrapperWorkItem.complete

### ISSUE THREAD

**Issue by jkasten2** [OneSignal/OneSignal-Android-SDK#644]
**Description:**
Reported Crashlytics crash, issue happening at `ccom.onesignal.JobIntentService$JobServiceEngineImpl$WrapperWorkItem.complete(JobIntentService.java:290)`.

Full stack trace below.
```Java
#0. Crashed: AsyncTask #4
       at android.os.AsyncTask$3.done(AsyncTask.java:353)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

--

Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:353)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

Caused by java.lang.IllegalArgumentException: Given work is not active: JobWorkItem{id=4 intent=Intent { (has extras) } dcount=1}
       at android.app.job.JobParameters.completeWork(JobParameters.java:233)
       at com.onesignal.JobIntentService$JobServiceEngineImpl$WrapperWorkItem.complete(JobIntentService.java:290)
       at com.onesignal.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:416)
       at com.onesignal.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:405)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
```

From investigation it seems this is happening when notifications are restored after reopening the app.

**Environment**
Almost all devices are Android 8.x.
Happening on OneSignal SDK 3.10.1 and 3.10.2.
`NotificationExtenderService` was not used.

**Steps to Reproduce Issue:**
Attempted to reproduce but was not able to, below is what was attempted;
1. Send 50 notifications
2. Force Kill the app.
3. Reopen it, see notifications are restored correctly
4. Tried redeploying app from Android Studio as well instead of step 2 + 3.
5. Tried with and without a big picture to a remote URL.

**Mitigating Issue**
While attempting to reproduce this issue I noticed that some apps could be trying to process restoring notifications that will end up being thrown away by the Android system due to some limits.
```logcat
E/NotificationService: Package enqueue rate is 10.56985. Shedding events. package=####
E/NotificationService: Package has already posted 50 notifications. Not showing more.  package=####
```
Restoring notifications can be slowed down as well as limiting to 50. This will at least prevent less jobs and possibly lower the chance of a race condition, if it is the cause here.

**Comment by jkasten2** [comment]
I haven't seen this bug come up in any recent builds, closing this issue.

**Comment by jkasten2** [comment]
This is due to a bug in the Android Support library which is a long running issue reported on Google's issue tracker.
https://issuetracker.google.com/u/1/issues/63622293?pli=1
The best approach is to handle this expectation as some other libraries such as evernote/android-job since Google hasn't solved this issue.

It seems there is a new API, [`WorkManager`](https://developer.android.com/topic/libraries/architecture/workmanager) that is probably best to migrate to at some point going forward however.

**Comment by jkasten2** [comment]
This is now fixed in the [3.10.8 release](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.10.8).

**Comment by atulgpt** [comment]
The issue is still happening it there any fix??
